src/scanner.ts

```diff
@@ -12,35 +12,48 @@ export interface ScanResult {
   ruleName: string;
   description: string;
   total: number;
-  ids: string[];
+  resources: {id: string}[];
 }
 
 export class Scanner {
-  private _rules: Rule[] = [];
```

Comment:
Without a rules property / data to go along with this class, it becomes kind of a named bucket of static functions, so there's no longer really any reason to new up a `Scanner`. Nothing necessarily wrong with that. But we're now making callers of this class create a RuleContext, do things with it, and then pass it back in to us. Feels less cohesive and like a step backwards imo - we should commit to "this is a useful class that does things for you" or "this is a purely functional interface and you chain things together to consume it"